src/main/java/appeng/fluids/parts/PartSharedFluidBus.java

```diff
@@ -0,0 +1,209 @@
+ * Copyright (c) 2013 - 2018, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.fluids.parts;
+
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.items.IItemHandler;
+
+import appeng.api.AEApi;
+import appeng.api.config.RedstoneMode;
+import appeng.api.config.Upgrades;
+import appeng.api.networking.ticking.IGridTickable;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.parts.IPartCollisionHelper;
+import appeng.api.storage.channels.IFluidStorageChannel;
+import appeng.api.util.AECableType;
+import appeng.core.sync.GuiBridge;
+import appeng.me.GridAccessException;
+import appeng.parts.automation.PartUpgradeable;
+import appeng.tile.inventory.AppEngInternalAEInventory;
+import appeng.util.Platform;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 30/04/2018
+ * @since rv6 30/04/2018
+ */
+public abstract class PartSharedFluidBus extends PartUpgradeable implements IGridTickable
+{
+
+	private final AppEngInternalAEInventory config = new AppEngInternalAEInventory( this, 9 );
+	private boolean lastRedstone;
+
+	public PartSharedFluidBus( ItemStack is )
+	{
+		super( is );
+	}
+
+	@Override
+	public void upgradesChanged()
+	{
+		this.updateState();
+	}
+
+	@Override
+	public void onNeighborChanged( IBlockAccess w, BlockPos pos, BlockPos neighbor )
+	{
+		this.updateState();
+		if( this.lastRedstone != this.getHost().hasRedstone( this.getSide() ) )
+		{
+			this.lastRedstone = !this.lastRedstone;
+			if( this.lastRedstone && this.getRSMode() == RedstoneMode.SIGNAL_PULSE )
+			{
+				this.doBusWork();
+			}
+		}
+	}
+
+	private void updateState()
+	{
+		try
+		{
+			if( !this.isSleeping() )
+			{
+				this.getProxy().getTick().wakeDevice( this.getProxy().getNode() );
+			}
+			else
+			{
+				this.getProxy().getTick().sleepDevice( this.getProxy().getNode() );
+			}
+		}
+		catch( final GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)